### PR TITLE
Use channels for non public content

### PR DESCRIPTION
**Pull request by annando**
By now, we only used public content from federated networks for the channels. With this change we now also use all content (including connector networks and private posts) that is available for the respective user.

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -1310,6 +1310,7 @@ CREATE TABLE IF NOT EXISTS `post-engagement` (
 	`media-type` tinyint NOT NULL DEFAULT 0 COMMENT 'Type of media in a bit array (1 = image, 2 = video, 4 = audio',
 	`language` varbinary(128) COMMENT 'Language information about this post',
 	`created` datetime COMMENT '',
+	`restricted` boolean NOT NULL DEFAULT '0' COMMENT 'If true, this post is either unlisted or not from a federated network',
```
Comment (MrPetovan) [inline database.sql:1313]: Why not name this field "unlisted"?

File: src/Model/Item.php [hunk 2]
```diff
@@ -1405,7 +1405,9 @@ public static function insert(array $item, int $notify = 0, bool $post_local = t
 			self::updateDisplayCache($posted_item['uri-id']);
 		}
 
-		Post\Engagement::storeFromItem($posted_item);
+		if ($inserted) {
```
Comment (MrPetovan) [inline src/Model/Item.php:1408]: Is it the only statement that needs to be inside this condition?

File: src/Model/Post.php [hunk 3]
```diff
@@ -36,10 +36,10 @@ class Post
 	 *
 	 * @param integer $uri_id
 	 * @param array   $fields
-	 * @return int    ID of inserted post
+	 * @return bool   Success of the insert process
```
Comment (MrPetovan) [inline src/Model/Post.php:39]: Why the change?

File: src/Model/Post/Engagement.php [hunk 4]
```diff
@@ -77,6 +65,15 @@ public static function storeFromItem(array $item)
 			$store = Contact::hasFollowers($parent['owner-id']);
 		}
 
+		if (!$store) {
+			$tagList = Relay::getSubscribedTags();
+			foreach (array_column(Tag::getByURIId($item['parent-uri-id'], [Tag::HASHTAG]), 'name') as $tag) {
+				if (in_array($tag, $tagList)) {
+					$store = true;
```
Comment (MrPetovan) [inline src/Model/Post/Engagement.php:72]: You can exit early from the `foreach` if the condition is met.
```suggestion
					$store = true;
					break;
```

File: src/Module/Contact/Profile.php [hunk 5]
```diff
@@ -340,13 +340,8 @@ protected function content(array $request = []): string
...
 
-		if (in_array($contact['network'], Protocol::FEDERATED)) {
-			$channel_settings_label = $this->t('Channel Settings');
-			$channel_frequency     = Contact\User::getChannelFrequency($contact['id'], $this->session->getLocalUserId());
-		} else {
-			$channel_settings_label = null;
-			$channel_frequency     = null;
-		}
+		$channel_settings_label = $this->t('Channel Settings');
+		$channel_frequency     = Contact\User::getChannelFrequency($contact['id'], $this->session->getLocalUserId());
```
Comment (MrPetovan) [inline src/Module/Contact/Profile.php]: Please move the `L10n` call to the template variable array below now that the variable can't be anything else.
```suggestion
		$channel_frequency = Contact\User::getChannelFrequency($contact['id'], $this->session->getLocalUserId());
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: database.sql (hunk 1 above)
Comment (annando) [reply database.sql:1313]: "unlisted" is a very specific term that is used for public posts that aren't that public. But this field also covers private posts and public posts from connector networks and rss/atom feeds.

File: src/Model/Item.php (hunk 2 above)
Comment (annando) [reply src/Model/Item.php:1408]: I haven't had a deep look at the code concerning this question.

File: src/Model/Post.php (hunk 3 above)
Comment (annando) [reply src/Model/Post.php:39]: The `Post` table has got no auto increment field, so this always returned `0`. Also the other `Post` related `insert` commands (for the other tables) also only return a boolean.

File: src/Module/Contact/Profile.php (hunk 5 above)
Comment (annando) [reply src/Module/Contact/Profile.php]: I don't understand what you mean here.

File: database.sql (hunk 1 above)
Comment (MrPetovan) [reply database.sql:1313]: Hmmm.

File: src/Module/Contact/Profile.php (hunk 5 above)
Comment (MrPetovan) [reply src/Module/Contact/Profile.php]: Please move the `$this->t('Channel Settings')` call to the current line 429 that reads:
```php
'$channel_settings_label' => $channel_settings_label,
```
instead of the variable, that you should remove as it doesn't have any other use now.

Comment (annando) [reply src/Module/Contact/Profile.php]: Done

**Review by MrPetovan — APPROVED** [review]
